src/analytics-dashboard/analytics-dashboard.service.ts

```diff
@@ -0,0 +1,514 @@
+/* eslint-disable no-console */
+import { Inject, Injectable } from "@nestjs/common";
+import {
+  CompanyModel,
+  JobModel,
+  OnCampusOfferModel,
+  ProgramModel,
+  RegistrationModel,
+  SalaryModel,
+  StudentModel,
+  UserModel,
+} from "src/db/models";
+import { ON_CAMPUS_OFFER_DAO, PROGRAM_DAO, REGISTRATIONS_DAO, STUDENT_DAO, USER_DAO } from "src/constants";
+import { Op, Sequelize } from "sequelize";
+import { CategoryEnum, GenderEnum } from "src/enums";
+
+@Injectable()
+export class AnalyticsDashboardService {
+  constructor(
+    @Inject(ON_CAMPUS_OFFER_DAO) private onCampusRepo: typeof OnCampusOfferModel,
+    @Inject(REGISTRATIONS_DAO) private registrationsRepo: typeof RegistrationModel,
+    @Inject(PROGRAM_DAO) private programRepo: typeof ProgramModel
+  ) {}
+
+  private calculateStatistics(ctcArray: number[]) {
+    if (ctcArray.length === 0) {
+      return {
+        highestPackage: 0,
+        lowestPackage: 0,
+        meanPackage: 0,
+        medianPackage: 0,
+        modePackage: 0,
+      };
+    }
+
+    const highestPackage = Math.max(...ctcArray);
+    const lowestPackage = Math.min(...ctcArray);
+    const meanPackage = ctcArray.reduce((acc, val) => acc + val, 0) / ctcArray.length;
+
+    ctcArray.sort((a, b) => a - b);
+    const mid = Math.floor(ctcArray.length / 2);
+    const medianPackage = ctcArray.length % 2 !== 0 ? ctcArray[mid] : (ctcArray[mid - 1] + ctcArray[mid]) / 2;
```

Comment:
we can maybe use https://mathjs.org/docs/reference/functions/median.html for all these. only a suggestion, but your call